File: frontend/src/types/course.ts
export interface Lesson {
  id: number;
  title: string;
}

export interface Module {
  id?: number;
  title: string;
  lessons: string[];
}
export interface Course {
  id: number;
  title: string;
  image: string; // Add this line
  category: string;
  description?: string;
  modules: Module[];
}

File: backend/Models/Course.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models
{
    public class Course
    {

        public int Id { get; set; }
        [Required]
        public string Title { get; set; } = string.Empty;
        public string Image { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public string? Description { get; set; }

        // Relationship: One Course has many Modules
        public List<Module> Modules { get; set; } = new();



    }
}
